komodo/lint_maturity.py

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import sys
+import yaml
+from packaging.version import Version, parse, InvalidVersion
+import warnings
+
+
+INVALID_TAGS = {
+    "a": ["Invalid"],
+    "b": ["a", "Invalid"],
+    "rc": ["a", "b", "Invalid"],
+    "stable": ["a", "b", "rc", "Invalid"],
+}
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(description=("Lint the maturity of packages."))
+    parser.add_argument(
+        "tag_exceptions",
+        type=lambda arg: arg
+        if os.path.isfile(arg)
+        else parser.error("{} is not a file".format(arg)),
+        nargs="+",
+        help="One file with exceptions for (a) release tag and (b) packages tag.",
+    )
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument(
+        "--release_file",
+        type=lambda arg: arg
+        if os.path.isfile(arg)
+        else parser.error("{} is not a valid file".format(arg)),
+        help="Release file",
+    )
+    group.add_argument(
+        "--release_folder",
+        type=lambda arg: arg
+        if os.path.isdir(arg)
+        else parser.error("{} is not a valid directory".format(arg)),
+        help="File with all package tags named as release version.",
+    )
+    return parser
+
+
+def get_release_type(version_string):
+    try:
+        version = Version(version_string)
+        release_type = "stable" if version.pre is None else version.pre[0]
+    except InvalidVersion:
+        release_type = "Invalid"
+    return release_type
+
+
+def get_release_info(file_to_lint, tag_exceptions_release):
```

Comment:
Great that you split your function up, now it is much easier to read. Another thing that would be great now that you´ve done that would be to write unit tests for these smaller functions. This function could for example be tested using `pytest parameterized`:
```
@pytest.mark.parametrize(
    "input_file,exceptions,clean_release,version,warning",
    [("/a/random/path/2020.02.a1-py27", [], "2020.02.a1-py27", "a", "")],
)
def test_release_info(input_file, exceptions, clean_release, version, warning):
    release_name, package_version, warning_msg = lint_maturity.get_release_info(
        input_file, exceptions
    )
    assert release_name == clean_release
    assert package_version == version
    assert warning_msg == warning
```
Then you can add different test cases to that.